tests/integration/test_implied_relationship_strategies.py

```diff
@@ -0,0 +1,170 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Test the various strategies for adding implied relationships."""
+
+import pytest
+
+from structurizr.model import InteractionStyle, Model
+from structurizr.model.implied_relationship_strategies import (
+    create_implied_relationships_unless_any_exist,
+    create_implied_relationships_unless_same_exists,
+    default_implied_relationship_strategy,
+)
+
+
+def test_by_default_model_doesnt_create_implied_relationships():
+    """Check if not set on Model, no implied relationships are added."""
+    model = Model()
+    assert model.implied_relationship_strategy is None
+
+    system1 = model.add_software_system(name="system1")
+    container1 = system1.add_container(name="container1", description="test")
+    system2 = model.add_software_system(name="system2")
+    container2 = system2.add_container(name="container2", description="test")
+
+    rel = container1.uses(container2, "Uses")
+
+    assert len(list(container1.get_relationships())) == 1
+    assert len(list(system1.get_relationships())) == 0
+    assert set(container1.get_relationships()) == {rel}
+    assert set(system1.get_relationships()) == set()
+
+
+def test_default_implied_relationship_strategy():
+    """Check that by default no implied relationships are added."""
+    model = Model(implied_relationship_strategy=default_implied_relationship_strategy)
+    system1 = model.add_software_system(name="system1")
+    container1 = system1.add_container(name="container1", description="test")
+    system2 = model.add_software_system(name="system2")
+    container2 = system2.add_container(name="container2", description="test")
+
+    rel = container1.uses(container2, "Uses")
+
+    assert len(list(container1.get_relationships())) == 1
+    assert len(list(system1.get_relationships())) == 0
+    assert set(container1.get_relationships()) == {rel}
+    assert set(system1.get_relationships()) == set()
+
+
+def test_create_implied_relationships_unless_any_exist():
+    """Check logic of create_implied_relationships_unless_any_exist."""
+    model = Model()
+    model.implied_relationship_strategy = create_implied_relationships_unless_any_exist
```

Comment:
```suggestion
    model = Model(implied_relationship_strategy=create_implied_relationships_unless_any_exist)
```

Reply:
Yeah, the problem here is hitting the line length limit.  Any chance we can bump that up a bit as I do find 88 to be difficult when trying not to abbreviate names?

Reply:
Actually, an alternative is that I alias the strategies when importing them to shorter names.  I'll give that a go instead.

Reply:
Hmmm, black makes the import aliases really ugly, but the main code looks better now.